src/presentation/http/router/note.test.ts

```diff
@@ -49,4 +52,70 @@ describe('Note API', () => {
       expect(body).toStrictEqual({ message: 'Note not found' });
     });
   });
-});
+
+  describe('GET note/:notePublicId ', () => {
+    test('Returns note by public id with 200 status', async () => {
+      const expectedStatus = 200;
+      
+      const expectedNote = {
+        'id': 2,
+        'publicId': 'Pq1T9vc23Q',
+        'creatorId': 1,
+        'content': null,
+        'createdAt': '2023-10-16T13:49:19.000Z',
+        'updatedAt': '2023-10-16T13:49:19.000Z',
+      };
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: '/note/Pq1T9vc23Q',
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      const body = response?.body !== undefined ? JSON.parse(response?.body) : {};
+
+      expect(body).toStrictEqual(expectedNote);
+    });
+
+    test('Returns 403 when public access is disabled in the note settings', async () => {
+      const expectedStatus = 403;
+
+      const notPublicNote = notes.find(note => {
+        const settings = noteSettings.find(ns => ns.note_id === note.id)
+        if(settings != undefined)
+        return settings.is_public === false
+      })
+
+      if(notPublicNote != undefined) {
```

Comment:
if mocked data are correct, we should not pass here, so it's better to case find result to Note instead since we know that it can't be undefined

Reply:
yes, and use casting instead